manifests/init.pp

```diff
@@ -44,11 +44,12 @@
   Enum['refpolicy', 'simple'] $default_builder                = 'simple',
 
   ### START Hiera Lookups ###
-  $boolean        = undef,
-  $fcontext       = undef,
-  $module         = undef,
-  $permissive     = undef,
-  $port           = undef,
+  Optional[Hash] $boolean         = lookup('selinux::boolean',         { 'default_value' => undef, 'merge' => 'hash' }),
```

Comment:
this does not work IMHO. `lookup()` will never be used because automatic parameter lookup will see selinux::boolean defined and then the default is not applicable. The defaults for the lookup need to be defined in hiera data itself. see https://puppet.com/docs/puppet/5.0/hiera_merging.html#configuring-merge-behavior-in-hiera-data

Reply:
Hmm. i think you're right. 
But i'm not sure what a good solution would be. 

Would lookup options defined in a module hiera be actually used if hiera finds the key in global/environment hiera? I would be surprised if it did.

Normally i would simply remove the merge options and leave that up to the user. But that might break backwards compatibility in this case.

Reply:
ok, forget what i wrote ;)
i've looked up the documentation and lookup_options is a special case for which puppet actually does a merge to cover this use-case we have here explicitly.

i'll add it to hiera